lib/galaxy/datatypes/qiime2.py

```diff
@@ -1,13 +1,28 @@
 import io
-import zipfile
+import ast
 import uuid as _uuid
+import zipfile
 
 import yaml
 
 from galaxy.datatypes.binary import CompressedZipArchive
 from galaxy.datatypes.metadata import MetadataElement
 
 
+def strip_properties(expression):
+    try:
+        expression_tree = ast.parse(expression)
+        PredicateRemover().visit(expression_tree)
+        reconstructer = ReconstructExpression()
+        reconstructer.visit(expression_tree)
+        return reconstructer.expression
+    # If we have any problems stripping properties just use the full expression
+    # this punts the error off to the GUI so if we error we do so there and not
```

Comment:
It makes the error `q2galaxy`'s problem. I don't think Galaxy would be any the wiser.